helm/managed/logging/elastic/values.yaml

```diff
@@ -21,6 +22,44 @@ elasticsearch:
       # Add default labels for the volumeClaimTemplate fo the StatefulSet
       #enabled: false
     #annotations: {}
+  extraEnvs:
+    # Increase WAIT_ES_READY_IN_SEC if get 'Cluster health check is unknown' message in log
+    - name: WAIT_ES_READY_IN_SEC
+      value: "120"
+  lifecycle:
+    postStart:
+      exec:
+        command:
+          - bash
+          - -c
+          - |
+            PIPELINE_NAME=hpccpipeline
+            ES_URL=http://localhost:9200
+            PIPELINE_REQ_PATH=_ingest/pipeline
+            [ -z ${WAIT_ES_READY_IN_SEC} ] && WAIT_ES_READY_IN_SEC=120
+            END_TIME=$(expr $(date +%s) \+ ${WAIT_ES_READY_IN_SEC})
+            START_FILE=/tmp/.es_start_file
+            while [[ $(date +%s) -le ${END_TIME} ]]; do
+              [ -f "${START_FILE}" ] && break
+              sleep 1
+            done
+            if [ -f "${START_FILE}" ]; then
+              echo 'Cluster is ready to add hpccpipeline'
+            else
+              echo 'Cluster health check timeout. Will try to add hpccpipeline anyway.'
+            fi
+            # Need wait elasticsearch container ready to test the hostname with ordinal
+            # Only add pipeline in the first elasticsearch master
+            if [ ${HOSTNAME##*-} -eq 0 ]; then
+              curl -X PUT "$ES_URL/$PIPELINE_REQ_PATH/$PIPELINE_NAME/" -H 'Content-Type: application/json' \
+                -d'{ "description": "Parses and formats HPCC Systems component log entries", "processors": [{ "grok": { "field": "message", "patterns": ["%{BASE16NUM:hpcc.log.sequence}\\s+%{HPCC_LOG_AUDIENCE:hpcc.log.audience}\\s+%{HPCC_LOG_CLASS:hpcc.log.class}\\s+%{TIMESTAMP_ISO8601:hpcc.log.timestamp}\\s+%{POSINT:hpcc.log.procid}\\s+%{POSINT:hpcc.log.threadid}\\s+%{HPCC_LOG_WUID:hpcc.log.jobid}\\s+%{QUOTEDSTRING:hpcc.log.message}"], "pattern_definitions": { "HPCC_LOG_WUID": "([A-Z][0-9]{8}-[0-9]{6})|(UNK)", "HPCC_LOG_CLASS": "DIS|ERR|WRN|INF|PRO|MET|UNK", "HPCC_LOG_AUDIENCE": "OPR|USR|PRG|AUD|UNK" } } }], "on_failure": [{ "set": { "field": "error.message", "value": "{{ _ingest.on_failure_message }}" } }] }'
+              if [ $? -eq ]; then
+                echo 'Successfully added hpccpipeline !'
+              else
+                echo 'Failed to add hpccpipeline.'
```

Comment:
Will the actual error be reported in this situation?

Reply:
Since postStart output will not in container's lod (kubectl logs) users can't see any stdout and stderr.  All they will experience when failed is the pod restarting